From 743ac0846de33fd4e3af03b5a94689027de9c479 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Mon, 6 Jun 2022 23:47:21 +0200
Subject: [PATCH] arm64: dts: qcom: cp01-c1: enable SDHCI controller

Enable the SDHCI controller for onboard uSD card slot.
CD and WP GPIO-s are present.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 arch/arm64/boot/dts/qcom/ipq6018-cp01-c1.dts | 25 ++++++++++++++++++++
 1 file changed, 25 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq6018-cp01-c1.dts
+++ b/arch/arm64/boot/dts/qcom/ipq6018-cp01-c1.dts
@@ -8,6 +8,7 @@
 /dts-v1/;
 
 #include "ipq6018.dtsi"
+#include <dt-bindings/gpio/gpio.h>
 
 / {
 	model = "Qualcomm Technologies, Inc. IPQ6018/AP-CP01-C1";
@@ -63,6 +64,20 @@
 		drive-strength = <8>;
 		bias-pull-down;
 	};
+
+	sdio_pins: sdio-state {
+		cd-pins {
+			pins = "gpio62";
+			function = "gpio";
+			bias-pull-up;
+		};
+
+		wp-pins {
+			pins = "gpio63";
+			function = "gpio";
+			bias-pull-down;
+		};
+	};
 };
 
 &qpic_bam {
@@ -81,6 +96,16 @@
 	};
 };
 
+&sdhci {
+	status = "okay";
+
+	pinctrl-0 = <&sdio_pins>;
+	pinctrl-names = "default";
+	cd-gpios = <&tlmm 62 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&tlmm 63 GPIO_ACTIVE_HIGH>;
+	bus-width = <4>;
+};
+
 &qusb_phy_1 {
 	status = "ok";
 };
